kythe/rust/extractor/src/lib.rs

```diff
@@ -29,75 +36,81 @@ use std::path::PathBuf;
 /// first element must be an empty string.
 /// The save_analysis JSON output file will be located at
 /// {output_dir}/save-analysis/{crate_name}.json
-pub fn generate_analysis(rustc_arguments: Vec<String>, output_dir: PathBuf) -> Result<(), String> {
+pub fn generate_analysis(rustc_arguments: Vec<String>, output_dir: PathBuf) -> Result<()> {
     let first_arg =
-        rustc_arguments.get(0).ok_or_else(|| "Arguments vector should not be empty".to_string())?;
+        rustc_arguments.get(0).ok_or_else(|| anyhow!("Arguments vector should not be empty"))?;
     if first_arg != &"".to_string() {
-        return Err("The first argument must be an empty string".into());
+        return Err(anyhow!("The first argument must be an empty string"));
     }
 
-    let mut callback_shim = CallbackShim::new(output_dir);
+    // Generate the configuration to parse the options
+    let mut options = getopts::Options::new();
+    for option in config::rustc_optgroups() {
+        (option.apply)(&mut options);
+    }
+    let matches = options.parse(&rustc_arguments[1..]).context("Failed to parse options")?;
```

Comment:
Are we going to need this same or very similar logic for this creating of the `config` and running the compiler for the indexer as well?

Reply:
No, because the indexer does not run the compiler